src/webserver/http.rs

```diff
@@ -434,25 +433,89 @@ async fn serve_file(
         })
 }
 
+/// Fallback handler for when a file could not be served
+///
+/// Recursively traverses upwards in the request's path, looking for a `404.sql` to call as the
+/// fallback handler. Fails if none is found, or if there is an issue while running the `404.sql`.
+async fn serve_fallback(
+    service_request: &mut ServiceRequest,
+    original_err: actix_web::Error,
+) -> actix_web::Result<HttpResponse> {
+    let catch_all = "404.sql";
+
+    let req_path = req_path(&service_request);
+    let mut fallback_path_candidate = req_path.clone().into_owned();
+    log::debug!("Trying to find a {catch_all:?} handler for {fallback_path_candidate:?}");
+
+    let app_state: &web::Data<AppState> = service_request.app_data().expect("app_state");
+
+    // Find the indeces of each char which follows a directroy separator (`/`). Also consider the 0
+    // index, as we also have to try to check the empty path, for a root dir `404.sql`.
+    for idx in req_path
+        .rmatch_indices('/')
+        .map(|(idx, _)| idx + 1)
+        .chain(std::iter::once(0))
+    {
+        // Remove the trailing substring behind the current `/`, and append `404.sql`.
+        fallback_path_candidate.truncate(idx);
+        fallback_path_candidate.push_str(&catch_all);
+
+        // Check if `maybe_fallback_path` actually exists, if not, skip to the next round (which
+        // will check `maybe_fallback_path`s parent directory for fallback handler).
+        let mabye_sql_path = PathBuf::from(&fallback_path_candidate);
+        let Ok(sql_file) = app_state
+            .sql_file_cache
+            .get_with_privilege(app_state, &mabye_sql_path, false)
+            .await
+        else {
```

Comment:
We have to check if the error here is indeed a file not found. If it's another error, then we can stop end return the error.

Reply:
This one does not spark joy, grr, `anyhow`.

Reply:
Done.

Reply:
Done for real.